--- a/dlls/nvapi/nvapi.c
+++ b/dlls/nvapi/nvapi.c
@@ -30,8 +30,9 @@
 #include "wine/list.h"
 #include "nvapi.h"
 #include "d3d9.h"
+#ifdef MESON_BUILD_D3D11
 #include "d3d11.h"
-
+#endif
 #include "wine/wined3d.h"
 
 WINE_DEFAULT_DEBUG_CHANNEL(nvapi);
@@ -683,6 +684,7 @@
     return NVAPI_OK;
 }
 
+#ifdef MESON_BUILD_D3D11
 static NvAPI_Status CDECL NvAPI_D3D11_SetDepthBoundsTest(IUnknown *pDeviceOrContext, NvU32 bEnable, float fMinDepth, float fMaxDepth)
 {
     struct wined3d_device *device;
@@ -761,6 +763,7 @@
 
     return NVAPI_OK;
 }
+#endif //MESON_BUILD_D3D11
 
 void* CDECL nvapi_QueryInterface(unsigned int offset)
 {
@@ -807,9 +810,11 @@
         {0x46fbeb03, NvAPI_GPU_GetPhysicalFrameBufferSize},
         {0x5a04b644, NvAPI_GPU_GetVirtualFrameBufferSize},
         {0xc7026a87, NvAPI_GPU_GetGpuCoreCount},
+#ifdef MESON_BUILD_D3D11
         {0x7aaf7a04, NvAPI_D3D11_SetDepthBoundsTest},
         {0x6a16d3a0, NvAPI_D3D11_CreateDevice},
         {0xbb939ee5, NvAPI_D3D11_CreateDeviceAndSwapChain},
+#endif
     };
     unsigned int i;
     TRACE("(%x)\n", offset);
